src/driver/mod.rs

```diff
@@ -26,29 +29,29 @@ use crate::{
     telemetry::metrics,
 };
 
-use self::engine_driver::EngineDriver;
+use self::engine_driver::{handle_attributes, ChainHead, EngineDriver};
 
-mod engine_driver;
+pub mod engine_driver;
 mod info;
 pub mod sequencing;
 mod types;
 pub use types::*;
 
 /// Driver is responsible for advancing the execution node by feeding
 /// the derived chain into the engine API
-pub struct Driver<E: Engine, S: sequencing::SequencingSource<E>> {
+pub struct Driver<E: Engine> {
     /// The derivation pipeline
     pipeline: Pipeline,
     /// The engine driver
-    engine_driver: EngineDriver<E>,
+    pub engine_driver: Arc<TokioRwLock<EngineDriver<E>>>,
     /// List of unfinalized L2 blocks with their epochs, L1 inclusions, and sequence numbers
     unfinalized_blocks: Vec<(BlockInfo, Epoch, u64, u64)>,
     /// Current finalized L1 block number
     finalized_l1_block_number: u64,
     /// List of unsafe blocks that have not been applied yet
     future_unsafe_blocks: Vec<ExecutionPayload>,
     /// State struct to keep track of global state
-    state: Arc<RwLock<State>>,
+    pub state: Arc<RwLock<State>>,
```

Comment:
can we also use tokio rwlock here? ref https://github.com/a16z/magi/issues/164

Reply:
it's okay as long as you don't hold the lock across await points (clippy checks this). async RwLocks have additional overhead unfortunately. I'll look into it separately